### DISCUSSION TITLE
Very different plotting speed on equal computers.

### DISCUSSION THREAD

**Discussion by alexey-sl** [Chia-Network/chia-blockchain#5187] [Support]
Have a strange issue here. 
Three computers.
First Ryzen 7 3700X 16Gb RAM 1Tb ssd kingstom m2 snvs1000g - linux
Second some pentium 8gb RAM with 2Tbb HDD WD purple - linux
Third Ryzen 7 3700X 16Gb RAM ssd apacer 480Gb as2280p4 - win10

1 plot creating time:
1. on 1st PC is somehow about 8 hours running on all 16 cores, thus core utilization is extremely low. 

2. Same plot creating on second weak PC is about 10-11 hours or so. Suddenly...

3. Some about 5 hours(4-40-4-50) on the same as #1 config but little faster ssd.

The question is - what is wrong with 1st PC. As it shows results in plot creating not much better of second pentium machine, that even not have a ssd at all. 

Thank you.

**Comment by alexey-sl** [comment]
For the moment noticed:
1. On the phase 1 very low CPU usage compared to #3 setup. 
2. When plotting is running - any parallel copy speed to ssd becomes very low, instead on the #3 setup when plotting, parallel copy speed to ssd falls too, but write speed drops not so dramatically.

**Comment by alexey-sl** [comment]
**Update. Plotting logs.**
1 - slow PC. 2- fast PC.

**_phase 1/4:_**
Computing table 1
_1. F1 complete, time: 201.449 seconds. CPU (131.94%)
2. F1 complete, time: 182.387 seconds. CPU (157.44%)_
Computing table 2
_1. Forward propagation table time: 638.883 seconds. CPU (460.750%)
2. Forward propagation table time: 749.939 seconds. CPU (536.740%)_
Computing table 3
_1. Forward propagation table time: 813.649 seconds. CPU (402.430%)
2. Forward propagation table time: 922.720 seconds. CPU (460.590%)_
Computing table 4
_1. Forward propagation table time: 2982.315 seconds. CPU (118.830%)
2. Forward propagation table time: 1186.616 seconds. CPU (391.240%)_
Computing table 5
_1. Forward propagation table time: 2718.978 seconds. CPU (125.500%)
2. Forward propagation table time: 1178.066 seconds. CPU (391.480%)_
Computing table 6
_1. Forward propagation table time: 2326.939 seconds. CPU (141.210%)
2. Forward propagation table time: 1017.668 seconds. CPU (435.290%)_
Computing table 7
_1. Forward propagation table time: 1648.812 seconds. CPU (143.190%)
2. Forward propagation table time: 806.835 seconds. CPU (532.050%)_

_1. Time for phase 1 = 11331.077 seconds. CPU (168.450%)
2. Time for phase 1 = 6044.272 seconds. CPU (439.080%)_

**_phase 2/4:_**
_1. Time for phase 2 = 5572.959 seconds. CPU (56.230%)
2. Time for phase 2 = 3118.445 seconds. CPU (95.960%)_

_**phase 3/4**_
_1. Time for phase 3 = 14339.569 seconds. CPU (45.480%)
2. Time for phase 3 = 7017.606 seconds. CPU (91.310%)_

 _**phase 4/4**_
_1. Time for phase 4 = 492.971 seconds. CPU (75.180%)
2. Time for phase 4 = 486.460 seconds. CPU (94.010%)_

**_Total_**
_1. Total time = 31736.579 seconds. CPU (91.730%) 
2. Total time = 16666.784 seconds. CPU (218.380%)_

As it goes from log compare, all is just fine till the start of "Computing table 4" of Phase 1/4.
The time and CPU usage differs slightly.
But after this moment the process begin to differ dramatically: CPU usage goes down and time taken for the process goes up. 

**Please advice where to dig.**

**Comment by Wojtasxj** [comment]
Have the same problem: 2 identical hardware machines (Ryzen9 5900X, 32GB RAM, Samsung 980Pro for temp)
Windows10 Pro -> 21000 seconds per plot
Ubuntu 20.04 -> 34000 seconds per plot
Unless reason for "slow ubuntu" is found, I revert back to Windows (perhaps I'll try another Centos first).
Anyone had the same UBUNTU problem?

**Reply by alexey-sl** [reply]
The question about windows/linux was investigated. 
I've set up a fresh debian installation on a removable USB HDD. Boot win10 machine from this HDD in linux. 
Mounted NTFS win10 partition with big_writes flag.
And plotted successfully in 5 hours. 
Regardless of fact that NTFS is not native to linux, it was approx 20 minutes longer of win10 result. Just because NTFS driver consumed 30-50% of one core. 
In total - I have same result win10/linux on same hardware. 
If i reformat partition from NTFS to linux native, thsn it will be even faster.

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
